WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -156,6 +237,50 @@ export default function RegionManager() {
           </Form>
         </Modal.Content>
       </Modal>
+      <Modal
+        onClose={() => {
+          closeModal();
+          setSelectedGroup(null);
+          setNewLeadDelegate(null);
+        }}
+        open={openModalType === 'newLeadDelegate'}
+      >
+        <Modal.Content>
+          <Form>
+            <Form.Field
+              label="New Lead Delegate"
+              control={WcaSearch}
+              value={newLeadDelegate}
+              onChange={(e, { value }) => setNewLeadDelegate(value)}
+              model="user"
+              multiple={false}
+            />
+            <Form.Button onClick={() => {
+              closeModal();
+              setSelectedGroup(null);
+              setNewLeadDelegate(null);
+            }}
+            >
+              Cancel
+            </Form.Button>
+            <Form.Button
+              disabled={!newLeadDelegate}
+              onClick={async () => {
+                closeModal();
+                setLoading(true);
+                await removeOldLeadDelegate(selectedGroup.lead_user);
+                await addNewLeadDelegate(newLeadDelegate, selectedGroup);
+                sync();
+                setSelectedGroup(null);
+                setNewLeadDelegate(null);
+                setLoading(false);
+              }}
```

Comment:
Please no async/await in React components. Just do whatever needs to be done in the `onSuccess` hook of the `useSaveData` action

Reply:
Okay, removed them and added the logic in backend which looks more ideal.

Reply:
Muuuuuuuch better now :heart_eyes: